csrc/ir/utils.h

```diff
@@ -66,13 +66,23 @@ struct MatmulInputs {
   Val* bias = nullptr;
   Val* alpha = nullptr;
   Val* beta = nullptr;
+  // Ordering of dimensions M,N,K in inputs.
+  // Determined based on position of iterdomains.
+  // For addmm/matmul ([M,K] x [K,N]): M=0, N=2, K=1
+  // For linear ([M,K] x [N,K]): M=0, N=1, K=2
+  // mma_dims_pos = {m_pos, n_pos, k_pos}
+  std::optional<std::vector<int>> mma_dims_pos = std::nullopt;
```

Comment:
Besides uninitialized, when is this nullopt?

Reply:
No other case. Only when the pattern matching fails or when the positions cannot be properly assigned to M,N,K iterdomains.

Reply:
Removed `std::optional`